library/spdm_crypt_lib/fips/libspdm_selftest_slhdsa.c

```diff
@@ -0,0 +1,133 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2025 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_crypt_lib.h"
+#include "internal/libspdm_common_lib.h"
+#include "internal/libspdm_fips_lib.h"
+#include "library/malloclib.h"
+
+#if LIBSPDM_FIPS_MODE
+
+/**
+ * slhdsa self_test
+ **/
+bool libspdm_fips_selftest_slhdsa(void *fips_selftest_context)
+{
+    bool result = true;
+
+#if LIBSPDM_SLH_DSA_SHA2_256S_SUPPORT
+    libspdm_fips_selftest_context_t *context = fips_selftest_context;
+    LIBSPDM_ASSERT(fips_selftest_context != NULL);
+
+    /* any test fail cause the FIPS fail*/
+    if (context->tested_algo != context->self_test_result) {
+        return false;
+    }
+
+    /* check if run before.*/
+    if ((context->tested_algo & LIBSPDM_FIPS_SELF_TEST_SLH_DSA) != 0) {
+        return true;
+    }
+
+    uint8_t *signature;
+    size_t sig_size;
+    void *slhdsa_context;
+
+    sig_size = 29792;
+    signature = allocate_zero_pool(sig_size);
```

Comment:
Please don't use `allocate()` function.

One possible option is to define a buffer in fips_context and let integrator define it (either in heap or in global data area)